simpleviewinc/keg-herkin
# Zen 546 gherkin editor

**Ticket**: [ZEN-546](https://jira.simpleviewtools.com/browse/ZEN-546)

## Context
 * We need to create the step definitions tab on keg-herkin
 * Create the container and component to display this content

## Goal
 * Display list of all available steps (code)
 * Clean up feature and definitions editor
 * Added helpers for writing feature files
   * Popup to match step definitions
   * See a list of all definitions

## Updates
**TODO**

## Testing

**Setup**
* Pull this pr => `keg herkin && keg pr 20`
* Then start it with `keg herkin start --config ./repos/example/herkin.config.js --from keg-herkin:zen-546-gherkin-editor`

**Test 1**
* Open http://herkin-zen-546-gherkin-editor.local.kegdev.xyz in a browser tab
* Then select the `example feature` from the test file drop down
  * [Looks like this](http://snpy.in/lbW6v0)
* Ensure it loads the Code editor tag with the `example feature` selected
* [Looks like this](http://snpy.in/Si7zQi)

**Test 2**
* On the left side should be the definitions editor
* At the top should be two tabs
  * Ensure the `Active` tab is selected
* Click on the `List` tab
  * Ensure it displays as list of all loaded step definitions
    * [Looks like this](http://snpy.in/FY9557)
* Click on each of the header rows for each type
  * Ensure the open and display only the types matching the header
  * * [Looks like this](http://snpy.in/ZMliNR)
* Hover over one of the step definitions
  * You should see `Copy to Clipboard` text displayed
    * * [Looks like this](http://snpy.in/sQdpSg)
  * Click on it, then check the console in your inspector
    * It should give you a message that the text has been copied
      * * [Looks like this](http://snpy.in/gkJbRV)
      * *NOTE* Eventually we'll want to add some type of toast notification

**Test 3**
* After doing **Test 2**, you should have a definitions matcher in your clipboard
* Select the feature editor on the left hand side
* Paste the definition matcher text into the feature editor next to one of the other steps
  * [Looks like this](http://snpy.in/jYf3gz)
  * Ensure it copies the text into the feature editor
    * [Looks like this](http://snpy.in/SWZ2jD)

**Test 4**
* Select the feature editor, and start typing the keyword for any of the step types
  * Step Types => `Given / When / Then`
* Ensure it shows a pop-up to auto fille the word
  * [Looks like this](http://snpy.in/9qK0se)
* Select the word from the pop up, then type a `space`, then `I`
  * [Looks like this](http://snpy.in/jmvYWa)
  * Ensure a pop-up shows, with a list of step definitions matching the currently typed text
* Try different variations and ensure it works as expected
